assembly/src/ast/parsers/io_ops.rs

```diff
@@ -282,31 +289,30 @@ fn parse_non_hex_param_with_constants_lookup<R: RangeBounds<u64>>(
 }
 
 /// Parses a single hexadecimal parameter into multiple values and returns an appropriate push
-/// instruction node.
+/// instruction node. Parameter is expected to be 64 character string representing the Word.
 ///
 /// # Errors
 /// Returns an error if:
-/// - The length of hex string is not even.
-/// - The length of hex string is not divisible by 16.
+/// - The length of hex string is not equal to 64.
 /// - If the string does not contain a valid hexadecimal value.
 /// - If the parsed value is greater than or equal to the field modulus.
 fn parse_long_hex_param(op: &Token, param_str: &str) -> Result<Node, ParsingError> {
     // handle error cases where the hex string is poorly formed
-    if param_str.len() % HEX_CHUNK_SIZE != 0 {
+    if param_str.len() != HEX_CHUNK_SIZE * 4 {
         // hex string doesn't contain a valid number of bytes
         return Err(ParsingError::invalid_param_with_reason(
             op,
             1,
             &format!(
-                "hex string '{param_str}' does not contain a number of characters multiple 16"
+                "long hex string '{param_str}' does not contain 64 characters required for the Word"
```

Comment:
minor nit: let's change this to: "long hex string '{hex_str}' must contain exactly 64 characters"